dev/gallery.tsx

```diff
@@ -76,33 +77,78 @@ const styles = StyleSheet.create({
 
 export function Gallery() {
     const ids = useUniqueIdWithMock();
+    const params = useMemo(
+        () => new URLSearchParams(window.location.search),
+        [],
```

Comment:
**question**: Does `window.location.search` need to be a dependency?  Otherwise won't the memo only be calculated once ever?

Reply:
No, otherwise it'd cause the component to reload every time the URL was updated in the `useEffect`. We only want it to be calculated once.